application_sandbox/passthrough/main.cpp

```diff
@@ -0,0 +1,338 @@
+      PassthroughFrameData* frame_data,
+      vulkan::VkCommandBuffer* initialization_buffer,
+      size_t frame_index) override {
+    frame_data->command_buffer_ =
+        containers::make_unique<vulkan::VkCommandBuffer>(
+            data_->root_allocator, app()->GetCommandBuffer());
+
+    ::VkImageView raw_view = color_view(frame_data);
+
+    // Create a framebuffer with depth and image attachments
+    VkFramebufferCreateInfo framebuffer_create_info{
+        VK_STRUCTURE_TYPE_FRAMEBUFFER_CREATE_INFO,  // sType
+        nullptr,                                    // pNext
+        0,                                          // flags
+        *render_pass_,                              // renderPass
+        1,                                          // attachmentCount
+        &raw_view,                                  // attachments
+        app()->swapchain().width(),                 // width
+        app()->swapchain().height(),                // height
+        1                                           // layers
+    };
+
+    ::VkFramebuffer raw_framebuffer;
+    app()->device()->vkCreateFramebuffer(
+        app()->device(), &framebuffer_create_info, nullptr, &raw_framebuffer);
+    frame_data->framebuffer_ = containers::make_unique<vulkan::VkFramebuffer>(
+        data_->root_allocator,
+        vulkan::VkFramebuffer(raw_framebuffer, nullptr, &app()->device()));
+
+    (*frame_data->command_buffer_)
+        ->vkBeginCommandBuffer((*frame_data->command_buffer_),
+                               &sample_application::kBeginCommandBuffer);
+    vulkan::VkCommandBuffer& cmdBuffer = (*frame_data->command_buffer_);
+
+    VkClearValue clear;
+    vulkan::MemoryClear(&clear);
+
+    VkRenderPassBeginInfo pass_begin = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO,  // sType
+        nullptr,                                   // pNext
+        *render_pass_,                             // renderPass
+        *frame_data->framebuffer_,                 // framebuffer
+        {{0, 0},
+         {app()->swapchain().width(),
+          app()->swapchain().height()}},  // renderArea
+        1,                                // clearValueCount
+        &clear                            // clears
+    };
+
+    // Create vertex buffers and index buffer
+    const auto vertices_buf_create_info = GetBufferCreateInfo(
+        sizeof(kVertices),
+        VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_VERTEX_BUFFER_BIT);
+    const auto uv_buf_create_info = GetBufferCreateInfo(
+        sizeof(kUV),
+        VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_VERTEX_BUFFER_BIT);
+    auto vertices_buf =
+        app()->CreateAndBindHostBuffer(&vertices_buf_create_info);
+    auto uv_buf = app()->CreateAndBindHostBuffer(&uv_buf_create_info);
+    const ::VkBuffer vertex_buffers[2] = {*vertices_buf, *uv_buf};
+    const ::VkDeviceSize vertex_buffer_offsets[2] = {0, 0};
+
+    FlushHostVisibleBuffer(&*vertices_buf, sizeof(kVertices),
```

Comment:
Sorry. You should be able to use FilHostVisibleBuffer. It should also handle flushing.